src/celestia/audiosession.h

```diff
@@ -0,0 +1,62 @@
+#pragma once
+
+#include <celcompat/filesystem.h>
+
+namespace celestia
+{
+
+namespace constants
+{
+const float minAudioPan = -1.0f;
+const float maxAudioPan = 1.0f;
+const float defaultAudioPan = 0.0f;
+const float minAudioVolume = 0.0f;
+const float maxAudioVolume = 1.0f;
+const float defaultAudioVolume = 1.0f;
+const int minAudioChannel = 0;
+const int defaultAudioChannel = 0;
+}
+
+class AudioSession
+{
+ public:
+    explicit AudioSession(const fs::path &path, float volume, float pan, bool loop, bool nopause);
+
+    AudioSession() = delete;
+    AudioSession(const AudioSession&) = delete;
+    AudioSession(AudioSession&&)      = delete;
+    AudioSession &operator=(const AudioSession&) = delete;
+    AudioSession &operator=(AudioSession&&) = delete;
+
+    virtual ~AudioSession() = default;
+    virtual bool play(double startTime = -1.0) = 0;
+    virtual bool isPlaying() const = 0;
+    virtual void stop() = 0;
+    virtual bool seek(double seconds) = 0;
+
+    void setVolume(float volume);
+    void setPan(float pan);
+    void setLoop(bool loop);
+    void setNoPause(bool nopause);
+
+    bool nopause() const { return _nopause; }
+
+ protected:
+    fs::path path() const { return _path; }
+    float volume() const { return _volume; }
+    float pan() const { return _pan; }
+    bool loop() const { return _loop; }
+
+    virtual void updateVolume() = 0;
+    virtual void updatePan() = 0;
+    virtual void updateLoop() = 0;
+
+ private:
+    fs::path _path;
```

Comment:
We have two styles for private variables naming: without any prefixes in the oldest code and with `m_` prefix in newer one. Could you use `m_` instead of just `_`?